File: Coherent-Moving-Cluster/PythonSource/Convoy.py
#!/usr/bin/env python
""" generated source for module Convoy """
from Cluster import Cluster
class Convoy(object):
    """ generated source for class Convoy """
    assigned = bool()
    startTime = float()
    endTime = float()
    lifetime = int()
    cluster = None

    def __init__(self, assigned=None, lifetime=None, cluster=None):
        """ generated source for method __init__ """
        if assigned:
            self.assigned = assigned
        else:
            self.assigned = False
        if lifetime:
            self.lifetime = lifetime
        else:
            self.lifetime = 1
        if cluster:
            self.cluster = cluster


    def isAssigned(self):
        """ generated source for method isAssigned """
        return self.assigned

    def setAssigned(self, assigned):
        """ generated source for method setAssigned """
        self.assigned = assigned

    def getStartTime(self):
        """ generated source for method getStartTime """
        return self.startTime

    def setStartTime(self, startTime):
        """ generated source for method setStartTime """
        self.startTime = startTime

    def getEndTime(self):
        """ generated source for method getEndTime """
        return self.endTime

    def setEndTime(self, endTime):
        """ generated source for method setEndTime """
        self.endTime = endTime

    def getLifetime(self):
        """ generated source for method getLifetime """
        return (self.endTime) - (self.startTime) + 1

    def setLifetime(self, lifetime):
        """ generated source for method setLifetime """
        self.lifetime = lifetime

    def toString(self):

        res = "Start_time: " + str(self.getStartTime()) + ",\tEnd_Time: " + str(self.getEndTime()) + "\t(" + str(self.getLifetime()) + ")\nobj_list: ";

        # Collections.sort(this);

        # res += str(self.cluster[0])

        for oid in self.cluster.oids:

            res += ", " + str(oid)


        res += '\n'

        return res

    def intersection(self,  c):
        """ generated source for method intersection """
        c1 = Convoy()
        c1.startTime = c.startTime
        c1.cluster = Cluster(c.cluster.cluster_id)
        tmp = c1
        i = 0
        while i < len(self.cluster.oids):
            j = 0
            while j < len(c.cluster.oids):
                if self.cluster.oids[i] == c.cluster.oids[j]:
                    tmp.cluster.oids.append(c.cluster.oids[j])
                    tmp.cluster.points.append(c.cluster.points[j])
                j += 1
            i += 1
        return tmp

File: Coherent-Moving-Cluster/JavaSource/coherent-moving-cluster/coherent-moving-cluster/src/main/Convoy.java
package main;

import java.util.ArrayList;
import java.util.Collections;

public class Convoy extends Obj_List{

    boolean assigned;
    double startTime, endTime;
    int lifetime;

    public Convoy( boolean assigned, int lifetime) {
        this.assigned = assigned;
        this.lifetime = lifetime;
    }

    public Convoy() {
        this(false, 1);
    }


    public Convoy(Cluster c) {

        this();
        super.addAll(c);

    }

    public boolean isAssigned() {return assigned;}
    public void setAssigned(boolean assigned) {this.assigned = assigned;}
    public double getStartTime() { return startTime; }
    public void setStartTime(double startTime) {this.startTime = startTime; }
    public double getEndTime() { return endTime; }
    public void setEndTime(double endTime) { this.endTime = endTime; }
    public int getLifetime() { return (int)endTime - (int)startTime + 1; }
    public void setLifetime(int lifetime) { this.lifetime = lifetime; }

    public Convoy intersection(Convoy c) {
        Convoy tmp = new Convoy();
        tmp.startTime = c.startTime;

        for (int i =0; i<this.size() ;i++ ){
            for(int j =0; j<c.size(); j++) {
                if(this.get(i).equals(c.get(j)))
                    tmp.add(c.get(j));
            }
        }

        return tmp;
    }

    @Override
    public String toString() {

        String res = "Start_time: " + getStartTime() + ",\tEnd_Time: " + getEndTime() +"\t(" + getLifetime()+")\nobj_list: ";

        Collections.sort(this);

        res+= this.get(0);

        for (int i = 1; i < this.size(); i++) {
            Integer oid = this.get(i);
            res += ", "+ oid;
        }

        res+='\n';

        return res;
    }
}
